src/utils/dateUtils.ts

```diff
@@ -66,3 +66,14 @@ export function isInFuture(datestring: string): boolean {
 export function isSameDate(first: Date, second: Date): boolean {
   return first.toISOString().slice(0, 10) == second.toISOString().slice(0, 10);
 }
+
+export function isValidDate(date: Date): boolean {
+  if (date && !isNaN(date.getTime())) {
+    return true;
+  } else {
+    return false;
+  }
+}
+export function endDateIsBeforeStartDate(first: Date, second: Date): boolean {
```

Comment:
I think this should just be called `dateIsBefore` or `dateIsAfter`, because this can be used for other things than start/end dates.